libraries/core/classes/AppCommand/index.js

```diff
@@ -1,169 +1,171 @@
+   * Sets the minimum required shopgate lib version for the command.
    * @param {string} libVersion The library version.
-   * @returns {AppCommand}
+   * @return {AppCommand}
    */
   setLibVersion(libVersion) {
-    this.libVersion = libVersion;
+    if (isValidVersion(libVersion)) {
+      this.libVersion = libVersion;
+    } else {
+      logger.error('Invalid lib version', libVersion);
+    }
+
     return this;
   }
 
   /**
-   * Dispatches the command into the app.
-   * @param {Object} [params=null] The command params.
+   * Logs the command to the console.
+   * @private
+   * @return {AppCommand}
    */
-  dispatch(params = null) {
-    if (params) {
-      this.command.p = params;
-    }
-
-    if (hasSGJavaScriptBridge()) {
-      try {
-        if ('dispatchCommandsForVersion' in SGJavascriptBridge) {
-          SGJavascriptBridge.dispatchCommandsForVersion([this.command], this.libVersion);
-        } else {
-          SGJavascriptBridge.dispatchCommandsStringForVersion(
-            JSON.stringify([this.command]),
-            this.libVersion
-          );
-        }
-
-        this.logCommand();
-      } catch (exception) {
-        logger.error(exception);
-      }
-    } else {
-      this.logCommand();
-
-      const devServerCommands = [
-        'sendPipelineRequest',
-        'sendHttpRequest',
-        'sendDataRequest',
-        'getWebStorageEntry',
-      ];
+  logCommand() {
+    if (this.log && !this.commandsWithoutLog.includes(this.name)) {
+      const title = `AppCommand %c${this.name}`;
 
-      // Only commands which represent a request are sent to the development server.
-      if (devServerCommands.includes(this.command.c)) {
-        this.sendDevCommand();
-      }
+      logGroup(title, this.params || {}, '#8e44ad');
     }
+
+    return this;
   }
 
   /**
-   * Logs the command to the console.
+   * Creates the command object which will be dispatched through the JavaScript bridge.
+   * @private
+   * @return {Object|null}
    */
-  logCommand() {
-    if (
-      !this.log ||
-      this.command.c === 'sendPipelineRequest' ||
-      this.command.c === 'sendHttpRequest' ||
-      this.command.c === 'getWebStorageEntry'
-    ) {
-      return;
+  buildCommand() {
+    let command = null;
+    if (this.name) {
+      command = {
+        c: this.name,
+        ...this.params && { p: this.params },
+      };
     }
 
-    const title = `AppCommand %c${this.command.c}`;
-
-    if (this.command.p) {
-      logGroup(title, this.command.p, '#8e44ad');
-    } else logGroup(title, {}, '#8e44ad');
+    return command;
   }
 
   /**
-   * Sends the command to the development API.
-   * @private
+   * Dispatches the command to the app.
+   * The returned promise will not be rejected for now in error cases to avoid the necessity
+   * of refactoring within existing code. But it resolves with FALSE in those cases.
+   * @param {Object} params The command params.
+   * @return {Promise}
```

Comment:
This function is not returning a promise anymore.

Reply:
It does. functions which are prefixed with `async` are transpiled to promises. If you would inspect it's return value you would see something like `Promise { <pending> }`.
see https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Statements/async_function

But i've changed the JSDoc return tag like suggested here http://usejsdoc.org/tags-async.html, so that the `resolve` type gets more visible.